### PR TITLE
Add fastlane support for screenshots

### PR DESCRIPTION
**Pull request by aubincleme**
Reference issue : #102 

This PR does the following : 

* Add a simple configuration for fastlane snapshots (that may need to be tuned)
* Allow the application to be started with the argument `debug` in order to:
  * Not load the camera in the scan view
  * Directly display the input field for entering a barcode manually in the same scan view
* Refactor and add some accessibility identifiers, allowing to directly access UI components during the snapshot process. This will also help when performing UI tests.
* Add a new build : OpenFoodFactsSnapshots for instrumenting the main application and for performing snapshots. The current build has the following properties:
  * It contains the same configuration as the one used by the Android app and mentioned by @teolemon here : https://github.com/openfoodfacts/openfoodfacts-ios/issues/102#issuecomment-515099993
  * It loads properly the app in any supported locale, takes a screenshot of a product, loads multiple products in the history, and takes a screenshot of the history

More work is needed to perform a search based on a product, take more screenshots of the ingredients, … However, there is some risk that the PR might become very big at some point (it’s already quite huge IMO).

### CODE REVIEW

**Comment by aubincleme** [comment]
One idea to improve the current changes would be to define a more specific argument to be passed to the application on startup : `debug` is a bit too generic.

**Comment by teolemon** [comment]
```
/Users/pierre/Desktop/openfoodfacts-ios-issue-102/UITests/ProductSearchUITests.swift:24:44: Type 'AccessibilityIdentifiers' has no member 'productSearchBar'
/Users/pierre/Desktop/openfoodfacts-ios-issue-102/UITests/ProductSearchUITests.swift:35:44: Type 'AccessibilityIdentifiers' has no member 'productSearchBar'
/Users/pierre/Desktop/openfoodfacts-ios-issue-102/UITests/ProductSearchUITests.swift:53:21: Type 'AccessibilityIdentifiers' has no member 'scanButton'
/Users/pierre/Desktop/openfoodfacts-ios-issue-102/UITests/ProductSearchUITests.swift:60:44: Type 'AccessibilityIdentifiers' has no member 'productSearchBar'
/Users/pierre/Desktop/openfoodfacts-ios-issue-102/UITests/ProductDetailUITests.swift:57:44: Type 'AccessibilityIdentifiers' has no member 'productSearchBar'
```

**Comment by aubincleme** [comment]
@teolemon the issue with OpenFoodFactsUITests not building because of AccessibilityIdentifiers should be fixed in 58f2742. However, I’ve not been able to build it myself to verify that it works fine.

**Comment by aubincleme** [comment]
Ok … silly me again, just found the script for the CI, so I guess that we use `fastlane test` … Updated 58f2742 into fd1d7e3 to remove one last occurrence of a now refactored `AccessibilityIdentifiers` variable.

So now, UI tests are not building, but it doesn’t seem to be due to the changes in this PR ^^

**Comment by teolemon** [comment]
@aubincleme yes, the build has been broken for quite some time

File: Sources/ViewControllers/Products/Search/ScannerViewController.swift [hunk 1]
```diff
@@ -63,13 +63,17 @@ class ScannerViewController: UIViewController, DataManagerClient {
...
         allergenAlertShown = false
 
-        checkCameraPermissions()
-        configureVideoView()
-        configureSession()
-        configureOverlay()
-        configureFlashView()
-        configureTapToFocus()
-
+        if !CommandLine.arguments.contains("debug") {
```
Comment (philippeauriach) [inline Sources/ViewControllers/Products/Search/ScannerViewController.swift]: "debug" should not be use, as it implies that the app is run with the debugger attached (which occurs for example automatically when running the project in dev mode). As shown in https://docs.fastlane.tools/getting-started/ios/screenshots/#advanced-snapshot, you can use
```if NSUserDefaults.standardUserDefaults().boolForKey("FASTLANE_SNAPSHOT") {
    // runtime check that we are in snapshot mode
}``` 
to check if you are inside a screenshot test

Comment (aubincleme) [reply Sources/ViewControllers/Products/Search/ScannerViewController.swift]: Thanks @philippeauriach ; the issue should be fixed as part of 0ea7c07.

Comment (aubincleme) [reply Sources/ViewControllers/Products/Search/ScannerViewController.swift]: One note : we might need to also have this custom behavior (for overriding the scanner, for example) in other UI tests in the future. When we’ll be there, we could add a helper to determine whether or not we are in a test environment.

**Comment by teolemon** [comment]
There's a merge conflict on this one

**Comment by aleene** [comment]
Merge conflict due to recent changes to master.

**Pull request merged**